includes/Core/Email_Reporting/Email_Report_Data_Section_Part.php

```diff
@@ -0,0 +1,256 @@
+	 * @param array|null  $date_range     Optional date range array with 'startDate' and 'endDate'.
+	 * @param string|null $dashboard_link Optional dashboard deeplink.
+	 *
+	 * @throws InvalidArgumentException When validation fails.
+	 */
+	public function __construct(
+		$section_key,
+		$title,
+		array $labels,
+		array $values,
+		$trends = null,
+		$date_range = null,
+		$dashboard_link = null
+	) {
+		$section_key = is_string( $section_key ) ? $section_key : '';
+		$title       = is_string( $title ) ? $title : '';
+
+		if ( '' === $section_key ) {
+			throw new InvalidArgumentException( 'section_key must be a non-empty string' );
+		}
+		if ( '' === $title ) {
+			throw new InvalidArgumentException( 'title must be a non-empty string' );
+		}
+
+		$this->labels = array_map( 'strval', $labels );
+		$this->values = array_map( 'strval', $values );
+
+		if ( ! is_array( $labels ) ) {
+			throw new InvalidArgumentException( 'labels must be an array' );
+		}
+		if ( ! is_array( $values ) ) {
+			throw new InvalidArgumentException( 'values must be an array' );
+		}
+
+		if ( null !== $trends ) {
+			if ( ! is_array( $trends ) ) {
+				throw new InvalidArgumentException( 'trends must be an array or null' );
+			}
+			$this->trends = array_map( 'strval', $trends );
+		} else {
+			$this->trends = null;
+		}
+
+		if ( null !== $date_range ) {
+			if ( ! is_array( $date_range ) ) {
+				throw new InvalidArgumentException( 'date_range must be an array or null' );
+			}
+			// Validate presence of keys if provided.
+			$start = isset( $date_range['startDate'] ) ? (string) $date_range['startDate'] : null;
+			$end   = isset( $date_range['endDate'] ) ? (string) $date_range['endDate'] : null;
+			if ( null === $start || null === $end ) {
+				throw new InvalidArgumentException( 'date_range must contain startDate and endDate' );
+			}
+			$compare_start_provided = array_key_exists( 'compareStartDate', $date_range );
+			$compare_end_provided   = array_key_exists( 'compareEndDate', $date_range );
+			if ( $compare_start_provided xor $compare_end_provided ) {
+				throw new InvalidArgumentException( 'date_range must contain both compareStartDate and compareEndDate when comparison dates are provided' );
+			}
+			$this->date_range = array(
+				'startDate' => $start,
+				'endDate'   => $end,
+			);
+			if ( $compare_start_provided && $compare_end_provided ) {
+				$this->date_range['compareStartDate'] = (string) $date_range['compareStartDate'];
+				$this->date_range['compareEndDate']   = (string) $date_range['compareEndDate'];
+			}
+		} else {
+			$this->date_range = null;
+		}
+
+		if ( null !== $dashboard_link && ! is_string( $dashboard_link ) ) {
+			throw new InvalidArgumentException( 'dashboard_link must be a string or null' );
+		}
+
+		$this->section_key    = $section_key;
+		$this->title          = $title;
+		$this->dashboard_link = $dashboard_link;
+	}
+
+	/**
+	 * Gets the section key.
+	 *
+	 * @return string
+	 */
```

Comment:
```suggestion
	/**
	 * Gets the section key.
	 *
	 * @since n.e.x.t
	 *
	 * @return string
	 */
```

Also for other getter methods bellow